### PR TITLE
Fixed FW upgrade sequence.

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@mellanox.com>

**- What I did**
Fixed FW upgrade sequence

**- How I did it**
Added logic to start FW upgrade on switch reboot after SONiC-To-SONiC update.
This will solve the issue when FW upgrade can be interrupted by swss restart.

**- How to verify it**
1. Perform SONiC-To-SONiC upgrade (FW versions should be different)
2. Reboot switch
3. Start parallel SSH session and restart swss

**- Description for the changelog**
* N/A

**- External dependencies**
1. https://github.com/Azure/sonic-utilities/pull/333

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        echo "Failed to burn FW: errno=${EXIT_CODE}"
+
+        if [[ ${FORCE_REBOOT} != "yes" ]]; then
+            echo "Reboot is interrupted: use -f|--force to override"
+            exit 1
+        fi
+    fi
+fi
+
+/sbin/reboot $@
```
Comment (qiluo-msft) [inline device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: exec ?

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                FORCE_REBOOT="yes"
+            ;;
+        esac
+        shift
+    done
+}
+
+ParseArguments $@
+
+if [[ "${CURRENT_SONIC_IMAGE}" != "${NEXT_SONIC_IMAGE}" ]]; then
```
Comment (qiluo-msft) [inline device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: Can you reuse code in https://github.com/Azure/sonic-utilities/blob/8819b4f7445f6389c3ceea1ee667993f2b586eba/scripts/fast-reboot#L39-L58

File: device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+FW_UPGRADE_SCRIPT="/usr/bin/mlnx-fw-upgrade.sh"
+
+NEXT_SONIC_IMAGE="$(sonic_installer list | grep "Next: " | cut -d ' ' -f 2)"
```
[the remark this answers is not in the document]
Comment (nazariig) [reply device/mellanox/x86_64-mlnx_msn2700-r0/platform_reboot]: @qiluo-msft This script is a part of reboot where we already have this check. Do we really need to duplicate it?

**Comment by qiluo-msft** [comment]
Retest this please

**Pull request merged**